app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Brand;
+use App\Models\Color;
+use App\Models\Product;
+use App\Models\Size;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    public function index()
+    {
+        $products = Product::with(['brand','color','size'])->paginate(10);
+    
+        return view('admin.pages.product.index', ['products' => $products]);
+    }
+    public function add()
+    {
+        $brands=Brand::get();
+        $colors = Color::get();
+        $sizes = Size::get();
```

Comment:
proper indentation & select required columns. 
do not forget select required columns